### PR TITLE
[coin] update to version 4.0.2

### LINKED ISSUE #35710: [coin] update to 4.0.2
### Library name

coin

### New version number

4.0.2

### Other information that may be useful (release notes, etc...)

https://github.com/coin3d/coin/releases/tag/v4.0.2

PR will be ready shortly.

### PR DESCRIPTION
**Pull request by ghost**
Fixes #35710

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version 
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

Note that: 
* The patch `disable-cpackd.patch` was removed, as `cpackd` appears to be controlled through a new cmake build option that is off by default.
* The patch `fix-typedef.patch` was removed, as its changes are already applied.

### CODE REVIEW

**Comment by carsten-grimm-at-ipolog** [comment]
The pipelines report the following error when building `coin` on linux:
```
-- Performing post-build validation
warning: There should be no absolute paths, such as the following, in an installed package:
  /mnt/vcpkg-ci/packages/coin_x64-linux
  /mnt/vcpkg-ci/installed
  /mnt/vcpkg-ci/buildtrees/coin
  /mnt/vcpkg-ci/downloads
Absolute paths were found in the following files:
  /mnt/vcpkg-ci/packages/coin_x64-linux/share/Coin/conf/coin-default.cfg
error: Found 1 post-build check problem(s). To submit these ports to curated catalogs, please first correct the portfile: /mnt/vss/_work/1/s/ports/coin/portfile.cmake
error: building coin:x64-linux failed with: POST_BUILD_CHECKS_FAILED

```
The error for macosx is similar.

I will build on WSL to see if I can find the offending absolute paths. Hints are welcome.

**Comment by carsten-grimm-at-ipolog** [comment]
It appears that the issue with the post build check appears to stem from the file `coin.cfg.cmake.in`, which contains 
```
# prefix is determined in coin-config, making the package "movable"
# prefix="@prefix@"
host="@host@"
exec_prefix="@exec_prefix@"
version="@COIN_VERSION@"
cflags="@COIN_EXTRA_CFLAGS@"
cppflags="@COIN_EXTRA_CPPFLAGS@"
cxxflags="@COIN_EXTRA_CXXFLAGS@"
ldflags="@COIN_EXTRA_LDFLAGS@"
libs="@CMAKE_INSTALL_FULL_LIBDIR@"
msvcrt="@COIN_MSVC_LIBC@"
datadir="@CMAKE_INSTALL_FULL_DATADIR@"
includedir="@CMAKE_INSTALL_FULL_INCLUDEDIR@"
frameworkdir="@frameworkdir@"
htmldir="@CMAKE_INSTALL_FULL_DOCDIR@"
compiler="@COIN_COMPILER@"
objext="@OBJEXT@"
have_3ds_import=@FEAT_HAVE_3DS@
have_vrml97=@FEAT_HAVE_VRML97@
have_sound=@FEAT_HAVE_SOUND@
have_superglu=@FEAT_HAVE_SUPERGLU@
have_threads=@FEAT_HAVE_THREADS@
have_threadsafe=@FEAT_HAVE_SAFETHREAD@
compiler_is_gcc=@ac_compiler_gnu@
```
I suspect that each occurrence of `CMAKE_INSTALL_FULL` is problematic.

Meanwhile the entries in `coin.cfg.in` appear to be benign and in line with the behavior from 4.0.0. I will try to redirect CMake to use that instead. Locally, this fixes the post build check.

File: ports/coin/fix-typedef.patch [hunk 1]
```diff
@@ -1,22 +0,0 @@
-diff --git a/include/Inventor/elements/SoGLLazyElement.h b/include/Inventor/elements/SoGLLazyElement.h
```
Comment (Neumann-A) [inline ports/coin/fix-typedef.patch:1]: Is this fixed upstream? I don't want to re-add this patch for clang-cl

Comment (Neumann-A) [reply ports/coin/fix-typedef.patch:1]: Yeah seems to be fixed.

File: ports/coin/portfile.cmake [hunk 2]
```diff
@@ -1,12 +1,11 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO Coin3D/coin
-    REF Coin-4.0.0
-    SHA512 8a0289cab3e02a7417022fe659ec30a2dd705b9bacb254e0269ada9155c76c6aea0285c475cd6e663f5d7f2b49e60244b16baac7188d57e3d7f8ab08d228f21f
+    REF v4.0.2
```
Comment (Neumann-A) [inline ports/coin/portfile.cmake:4]:
```suggestion
    REF v${VERSION}
```

File: ports/coin/portfile.cmake [hunk 3]
```diff
@@ -32,7 +31,7 @@ vcpkg_cmake_configure(
 
 vcpkg_cmake_install()
 vcpkg_copy_pdbs()
-vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/Coin-4.0.0)
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/Coin-4.0.2)
```
Comment (Neumann-A) [inline ports/coin/portfile.cmake:34]:
```suggestion
vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/Coin-${VERSION})

```

File: ports/coin/fix-typedef.patch (hunk 1 above)
Comment (carsten-grimm-at-ipolog) [reply ports/coin/fix-typedef.patch:1]: Yes, I double checked before removing the patch.

**Review by FrankXie05 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**